### PR TITLE
Support when PYTHONIOENCODING is set to ascii

### LINKED ISSUE #15: maxdiff doesn't seem robust to different text encoding settings
Hi Mattijs,

first try with the very welcome maxdiff, I run into this problem:
```
%python3 ~/src/maxdevtools-ableton/maxdiff/amxd_textconv.py cocavs-image.amxd
Traceback (most recent call last):
  File "/Users/schwarz/src/maxdevtools-ableton/maxdiff/amxd_textconv.py", line 106, in <module>
    main()
  File "/Users/schwarz/src/maxdevtools-ableton/maxdiff/amxd_textconv.py", line 99, in main
    print(result)
UnicodeEncodeError: 'latin-1' codec can't encode character '\u2013' in position 150942: ordinal not in range(256)
```
I need to do `setenv PYTHONIOENCODING UTF-8` to make it work. Maybe my setup is weird, but can't the encoding be enforced by the scripts themselves? Best!

### PR DESCRIPTION
**Pull request by MattijsKneppers**
This fixes the issue that the scripts throw an error if the user has `PYTHONIOENCODING` set to `ascii` on their system.

Fixes #15

### CODE REVIEW

File: maxdiff/print_unicode.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Module that shares a function to gracefully handle printing unicode characters 
+if the user has the PYTHONIOENCODING environment vairable set to 'ascii'. 
+In that case, replace any characters that can't be encoded with ?.
+"""
+
+
+def print_unicode_string(s):
+    try:
+        encoded_string = s.encode("utf-8", "replace")
+        print(s)
```
Comment (jamesb93) [inline maxdiff/print_unicode.py:11]: Don't we want to return the now encoded string? Not the original?

File: maxdiff/tests/test_baselines/Test.amxd.txt [hunk 2]
```diff
@@ -21,6 +20,7 @@ project:
 		data:
 			collContent.txt
 ----------- objects -----------
+[message 😋] fontsize: 20.0
```
Comment (jamesb93) [inline maxdiff/tests/test_baselines/Test.amxd.txt:23]: Should we test that something like this is properly patch printed if the encoding is configured to something other than UTF-8 in the environment?

File: maxdiff/print_unicode.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    except UnicodeEncodeError:
+        encoded_string = s.encode("ascii", "replace")
```
Comment (jamesb93) [inline maxdiff/print_unicode.py:13]: What happens here if encoding fails? Are we also happy with ? for any character that it needs to replace?

Essentially, does this cover all the bases of an encoding being not UTF-8?

**Review by jamesb93 — COMMENTED** [review]
See my comments, just a few small things. Otherwise the general solution is fine, I think.

File: maxdiff/print_unicode.py (hunk 1 above)
Comment (MattijsKneppers) [reply maxdiff/print_unicode.py:11]: Ah you are absolutely right, the encode statement isn't needed there. The try already fails if we try to print s when it contains non-ascii characters and PYTHONIOENCODING is set to ascii. I'll fix this.

File: maxdiff/tests/test_baselines/Test.amxd.txt (hunk 2 above)
Comment (MattijsKneppers) [reply maxdiff/tests/test_baselines/Test.amxd.txt:23]: I think we should ideally, but I don't know how :p I'm ok to leave it like this for now and fix this when an idea comes up.

File: maxdiff/print_unicode.py (hunk 3 above)
Comment (MattijsKneppers) [reply maxdiff/print_unicode.py:13]: I agree, if the change in a patch would only be to use a different unicode character, we wouldn't see a diff with this approach. 

I'll replace the error handling flag to `backslashreplace`, which should convert UTF-8 characters to ascii code; e.g. the one used in the test patch converts to `\U0001f60b`.

**Review by jby-ableton — APPROVED** [review]
Looks great!

**Pull request merged**